Kubeadm unit tests pkg node

dmmcquay opened the pull request:
Added unit tests for the kubeadm/app/node package testing functionality of bootstrap.go, csr.go, and discovery.go. 

This PR is part of the ongoing effort to add tests (#35025)

/cc @pires @jbeda

<!-- Reviewable:start -->

---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36106)
<!-- Reviewable:end -->

dmmcquay committed:
kubeadm: unit tests for app/node/ pkg

pires commented:
@k8s-bot ok to test

pires commented:
The tests are passing but something happened with the last check:
```
W1103 11:20:09.036] + ./cluster/log-dump.sh _artifacts
I1103 11:20:09.126] Dumping master and node logs to _artifacts
I1103 11:20:10.922] Master not detected. Is the cluster up?
I1103 11:20:11.577] Nodes not detected. Is the cluster up?
W1103 11:20:11.578] + [[ 1 -eq 124 ]]
```

dmmcquay commented:
@pires Try running the test again?

pires commented:
@k8s-bot gci gce e2e test this

pires commented:
This LGTM. Can you squash into single commit, please?

pires commented:
@dmmcquay builds pass but can you squash, please?

dmmcquay commented:
@pires done.

pires commented:
/lgtm

pires commented:
Can you help @luxas?

dmmcquay commented:
@pires I think you just add the release-note-none label and remove the release-not-lable-needed.

pires commented:
Errmm I don't have the permission to do that..

luxas commented:
Done. However, it won't merge until the queue to master opens some time later this month

rmmh commented:
@k8s-bot unit test this

(I think this is making a particular batch fail)

dmmcquay committed:
kubeadm: updated pkg node tests to correct types

updated pkg "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset" to "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_5" because the type that bootstrap.go:126 checkAPIEndpoint requires changed as well as *internalversion.CertificatesClient in csr_test.go:69

pires commented:
/lgtm

dmmcquay commented:
@luxas looks like a flake, mind retesting?

luxas commented:
@k8s-bot gci gce e2e test this
